UploadDaemon/UploadDaemon.cs

```diff
@@ -81,27 +67,55 @@ private static bool IsAlreadyRunning()
             return Process.GetProcessesByName(current.ProcessName).Where(process => process.Id != current.Id).Any();
         }
 
-        public UploadDaemon(Config config)
+        public UploadDaemon()
         {
             fileSystem = new FileSystem();
+            uploadTask = new UploadTask(fileSystem, new UploadFactory(), new LineCoverageSynthesizer());
+        }
 
-            this.config = config;
+        private Config ReadConfig()
+        {
+            Config config;
+            try
+            {
+                logger.Debug("Reading config from {configFile}", Config.ConfigFilePath);
+                config = Config.ReadFromCentralConfigFile();
+            }
+            catch (Exception e)
+            {
+                logger.Error(e, "Failed to read config file {configPath}", Config.ConfigFilePath);
+                return null;
```

Comment:
I would move the exception handling to the callsite, to avoid passing around null. A method named ReadConfig may throw if reading fails imho.